presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -760,10 +794,19 @@ public void onFailure(Throwable throwable)
         return true;
     }
 
+    // TODO: Simplify the commit logic of the transaction manager.
```

Comment:
what does this todo for exactly?

Reply:
Previously transaction manager makes commit logic into two different paths: read-only and the rest. For read-only queries, the commit result is a list of commit handles; for the rest the commit result is just single commit handle. We should be unify both of them into a list of commit handles.